teps/0092-scheduling-timeout.md

```diff
@@ -17,361 +22,470 @@ authors:
-    what `scheduling` means in that environment. The recommendation would be to stop the timer once the execution of the
-    Run is ready to begin.
-
-## Drawbacks
-
-See [risks and mitigations](#risks-and-mitigations).
+This is an evaluation of the above proposal against the [Tekton Design Principles][design-priciples].
+
+#### Reusability:
+
+Since this is being added as a runtime feature, `Pipelines` and `Tasks` will be just as reusable as they were before the feature. If we need to specify these new timeouts at the level of `Tasks` within a `Pipeline`, this would make the `Pipelines` slightly less reusable since we would be potentially mixing environment specific configuration into the `Pipeline` definition.
+
+#### Simplicity
+* This does make the timeout logic more complex, and as mentioned in [risks and mitigations](#risks-and-mitigations) this continues the trend of adding more and more complex timeout logic. One upside is that this is only in the runtime configuration and not (yet) the Pipeline or Task authoring time.
+* This follows the same trend as the recent addition of [finally timeout configuration][finally-timeout]
+* As we add more timeout options it may become more difficult to reason about which timeout is applying when
+* It is debatable that this feature is absolutely necessary; there are use cases to support it however it could potentially not be widely used and would cost additional cognitive overhead for all users trying to reason through the timeout logic
+
+#### Flexibility
+* This proposal does not involve bringing in any new dependencies, coupling Pipelines to any new tools, or introducing templating logic
+* Again implementing this very specific timeout configuration in our API does open the door for adding more of this configuration and the proposal is not very flexible in that it adapting it to new timeout cases will mean adding new config options and explaining in detail how they relate to the existing ones
+
+#### Conformance
+
+* Likely we would include this timeout configuration in our conformance surface for `PipelineRuns` and `TaskRuns`.
+* If something other than Kubernetes was used to back the API, the implementors could make their own decisions about what `scheduling` means in that environment. The recommendation would be to stop the timer once the execution of the `TaskRun` is ready to begin.
+
+
+#### Performance
+
+* Additional timeouts to consider will mean additional `TaskRuns` queued for later reconciling within the controller.
+* This would mean at most 2x the number of timeouts for each individual `TaskRun`.
 
 ## Alternatives
 
-* [Including image pull and init container execution in the timeout](#continue-timeout-until-running)
-* Configure this timeout at the controller level as well
+### Configuration Levels
+
+* Configure these timeouts at the controller level as well
   * Pro: easy to set and forget in an environment that is always resource constrained
```

Comment:
It looks like this alternative is the same as the alternative "configure at the controller level instead", can their pros and cons be combined?